File: arvore/avl/Arborizavel.java
package avl;

public interface Arborizavel<T> {

    NoTriplo<T> getRaiz();
    void inserir(T dado);
    T apagar(T dado);
    boolean existe(T dado);
    void limpar();
    String imprimirPreOrdem();
    String imprimirEmOrdem();
    String imprimirPosOrdem();
}

File: arvore/Kotlin/ABP Genérica/Arborizavel.kt
interface Arborizavel<T> {
    fun getRaiz(): NoTriplo<T>?
    fun inserir(dado: T?)
    fun apagar(dado: T?): Boolean
    fun existe(dado: T?): Boolean
    fun buscar(dado: T): NoTriplo<T>?
    fun limpar()
    fun imprimirPreOrdem(): String
    fun imprimirEmOrdem(): String
    fun imprimirPosOrdem(): String
}